app/scripts/views/filelist.js

```diff
@@ -41,43 +33,28 @@ export default React.createClass({
           </tr>
         </thead>
         <tbody>
-        {files ? files.map(function (file) {
-          if (typeof file === 'string') file = { id: file }
-
-          var type = '?'
-          if (file.name) {
-            var lastDot = file.name.lastIndexOf('.')
-            if (lastDot !== -1) type = file.name.substr(lastDot + 1, 4).toUpperCase()
+        {files ? files.map(file => {
```

Comment:
Alright sounds good. Is this a function you'd want to define on the class or just inside the render method? I don't see much of a reason to define it on the class.